tests/test_definition.py

```diff
@@ -78,18 +78,33 @@ def test_to_excel(simple_definition, tmpdir):
 
     simple_definition.to_excel(file)
 
-    obs = pd.read_excel(file)
+    obs = pd.read_excel(file, sheet_name="variable")
     exp = pd.read_excel(TEST_DATA_DIR / "excel_io" / "validation_nc.xlsx")
     pd.testing.assert_frame_equal(obs, exp)
 
 
+def test_to_excel_with_external_repo(tmpdir):
+    """Check writing a DataStructureDefinition with an external repo to file"""
+    file = tmpdir / "testing_export.xlsx"
+
+    dsd = DataStructureDefinition(TEST_DATA_DIR / "general-config" / "definitions")
+    dsd.to_excel(file)
+
+    obs = pd.ExcelFile(file)
```

Comment:
Although not a critical as in the pyam issue iamconsortium/pyam#817, I'd also use a context manager here to make sure the file is properly closed at the end of the test:
```suggestion
    with pd.ExcelFile(file) as obs:
```

Reply:
Implemented, thanks for the suggestion